### DISCUSSION TITLE
classes/class_checkbox

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#48] [User-contributed notes]
# classes/class_checkbox

Inherits: Button< BaseButton< Control< CanvasItem< Node< Object A button that represents a binary choice. Description: CheckBox allows the user to choose one of only two possible options. It's simi...

https://docs.godotengine.org/en/stable/classes/class_checkbox.html

**Comment by mouseroot** [comment]
No Documentation on is_pressed to determine the value of the checkbox true for checked, false if not, took quite a bit of fishing to find it I think it should be in the docs as it's important to know the value of it

**Reply by Calinou** [reply]
The `button_pressed` property is part of the BaseButton class, which CheckBox inherits from. This is why it doesn't appear in the CheckBox class reference (to avoid redundancy).

The description at the top of the CheckBox class states this:

> See also [BaseButton](https://docs.godotengine.org/en/stable/classes/class_basebutton.html#class-basebutton) which contains common properties and methods associated with this node.

**Comment by harrypujols** [comment]
The checkbox button textures are missing the focus and hover states for both check and unchecked. It's a pain to have to program this every time I use this node.

**Reply by Calinou** [reply]
Please open a [proposal](https://github.com/godotengine/godot-proposals) for this.
